presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -216,6 +217,159 @@ else if (rescaleFactor > 0) {
         }
     }
 
+    public static Slice add(Slice left, Slice right)
+    {
+        Slice result = unscaledDecimal();
+        add(left, right, result);
+        return result;
+    }
+
+    public static void add(Slice left, Slice right, Slice result)
+    {
+        long overflow = addReturnOverflow(left, right, result);
+        if (overflow != 0) {
+            throwOverflowException();
+        }
+    }
+
+    /**
+     * Instead of throwing overflow exception, this function returns:
+     * 0 when there was no overflow
+     * +1 when there was overflow
+     * -1 when there was underflow
+     */
+    public static long addReturnOverflow(Slice left, Slice right, Slice result)
```

Comment:
Fair enough, let's keep the nice name for the common operations (my suggestion was modeled after Math.addExact and friends).

However, I'd rename the method to `addWithOverflow`